src/app/task-list/task-list.component.ts

```diff
@@ -42,18 +41,16 @@ export class TaskListComponent implements OnInit {
 
     this.taskService.deleteATask(taskId)
 
-      .pipe(
-        takeUntil(this.destroy$),
-        switchMap(() => this.taskService.getAllTasks()))
-
+      .pipe(takeUntil(this.destroy$))
       .subscribe({
-        next: (updatedTasks: Task[]) => {
-          this.tasks = updatedTasks;
+        next: () => {
+          this.tasks = this.tasks.filter(task => task._id !== taskId);
```

Comment:
the new tasks should already be filtered after you delete them in the backend, as soon as you trigger the delete function in the BE you should be returning the latest list of tasks (without the task you just deleted)